### DISCUSSION TITLE
tv4play och undertexter.

### DISCUSSION THREAD

**Discussion by DuckRandom** [stefansundin/privatkopiera#163] [General]
Jag tänkte bara nämna, ifall det är fler än jag som inte får med undertexter på tv4play via privatkopiera. 
(Också som en liten anteckning till mig själv nästa gång jag glömt hur jag gjorde :P)

1. Hoppa in på adressen. Som exempel kör vi Mandelmanns: https://www.tv4play.se/program/mandelmanns-g%C3%A5rd/del-5-s%C3%A4song-7/13786560
2. Kopiera m3u8-adressen från kommandorutan i privatkopiera: https://vod.streaming.a2d.tv/88d662f5-fe19-49c2-8aaf-13c1302fcbcf/57ba80a0-a623-11ed-a86a-198f8c1dc419_20396367.ism/.m3u8
3. Kör sedan yt-dlp --all-subs --convert-subs srt "https://vod.streaming.a2d.tv/88d662f5-fe19-49c2-8aaf-13c1302fcbcf/57ba80a0-a623-11ed-a86a-198f8c1dc419_20396367.ism/.m3u8"

Som vanligt så är mina lösningar kanske inte dom mest smidiga, men det funkar :)
Andra/smidigare tips är naturligtvis mycket välkomna.

**Comment by IronWalkers** [comment]
Var klistrar du in det i punkt 3? Gör jag det via CMD i mappen där ffmpeg finns får jag att det är felaktigt kommando (yt-dlp). 

Lägger jag till ffmpeg.exe innan för att få det som kommando liknande: ffmpeg.exe yt-dlp --all-subs --convert-subs srt "https://vod.streaming.a2d.tv/73ef8fa0-8b7b-4be4-a39f-1ce44beb20f7/b0195cc0-382d-11ee-b004-3bf3c37f20ca_20576705.ism/b0195cc0-382d-11ee-b004-3bf3c37f20ca_20576705-video=128000.m3u8"
blir det så här.
Unrecognized option '-all-subs'.
Error splitting the argument list: Option not found

**Reply by DuckRandom** [reply]
Usch, förlåt för sent svar.. Jag har inte varit inne på github på ett bra tag.
yt-dlp är ett annat program som är en fork av det äldre youtube-dl, som används för att ladda ner från (mestadels) youtube.
Du hittar all information här https://www.reddit.com/r/youtubedl/wiki/index

**Reply by A2D288** [reply]
Jag laddade ned yt-dlp och följde dina instruktioner ovan. I min Privatkopiera-länk finns två .m3u8-adresser - en för ljud och en för video. 

Oavsett vilken jag försöker med får följande info från yt-dlp: 
_[info] There are no subtitles for the requested languages
[SubtitlesConvertor] There aren't any subtitles to convert_

Här är min Privatkopiera-länk för första avsnittet av serien Konflikt på TV4 Play: 

`ffmpeg -i "https://vod.streaming.a2d.tv/902bdc72-425e-4bbc-892b-3170d84eda4f/6c86b630-21f4-11f0-b188-f56ceaf64f7c_20939410.ism/6c86b630-21f4-11f0-b188-f56ceaf64f7c_20939410-video=6500000.m3u8" -i "https://vod.streaming.a2d.tv/902bdc72-425e-4bbc-892b-3170d84eda4f/6c86b630-21f4-11f0-b188-f56ceaf64f7c_20939410.ism/6c86b630-21f4-11f0-b188-f56ceaf64f7c_20939410-audio=128000.m3u8" -c:v copy -c:a copy -c:s mov_text "C:\Users\***\Desktop\Invasion.mp4"`

Har du några råd?

**Reply by A2D288** [reply]
Jag missade visst att man skulle klippa lite i adressen. När jag gjorde exakt som det står i dina instruktioner (och tog bort det som står mellan ".ism/" och ".m3u8" i adressen) funkade det perfekt. Tack!

**Reply by DuckRandom** [reply]
Om du använder -o "namn på program" efteråt så blir det ännu smidigare:
Från exemplet ovan:  yt-dlp --all-subs --convert-subs srt https://vod.streaming.a2d.tv/88d662f5-fe19-49c2-8aaf-13c1302fcbcf/57ba80a0-a623-11ed-a86a-198f8c1dc419_20396367.ism/.m3u8 -o "Mandelmann s07e05"
